qemu/tests/migration_multi_host_with_stress.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from autotest.client.shared import utils, error
+
+from virttest import utils_test
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    KVM multi-host migration test:
+
+    Migration execution progress is described in documentation
+    for migrate method in class MultihostMigration.
+    steps:
+        1) boot vm and login, load stress in guest
+        2) do migration
+        3) wait for a while and stop stress
+        4) login guest when migrate done
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    preprocess_env = params.get("preprocess_env", "yes") == "yes"
+    mig_protocol = params.get("mig_protocol", "tcp")
+    mig_type = utils_test.qemu.MultihostMigration
+    if mig_protocol == "fd":
+        mig_type = utils_test.qemu.MultihostMigrationFd
+    if mig_protocol == "exec":
+        mig_type = utils_test.qemu.MultihostMigrationExec
+    if "rdma" in mig_protocol:
+        mig_type = utils_test.qemu.MultihostMigrationRdma
+
+    class TestMultihostMigration(mig_type):
+
+        def __init__(self, test, params, env):
+            super(TestMultihostMigration, self).__init__(test, params, env,
+                                                         preprocess_env)
+            self.srchost = self.params.get("hosts")[0]
+            self.dsthost = self.params.get("hosts")[1]
+            self.vm = params["vms"].split()[0]
+            self.login_timeout = int(params.get("login_timeout", 360))
+
+        def do_stress(self, session):
+            stress_cmd = params.get("stress_cmd")
+            s = session.cmd_status(stress_cmd, timeout=360)
+            if s != 0:
+                raise error.TestError("Failed to do stress in guest.")
+
+        def migration_scenario(self):
+            @error.context_aware
+            def worker(mig_data):
+                logging.info("Try to login guest before migration test.")
+                vm = env.get_vm(params["main_vm"])
+                session = vm.wait_for_login(timeout=self.login_timeout)
+
+                error.context("Do stress test before migration.", logging.info)
+                if params.get("os_type") == "linux":
+                    bg = utils.InterruptedThread(self.do_stress, (session,))
+                else:
+                    bg = utils.InterruptedThread(utils_test.run_virt_sub_test,
+                                                 args=(test, params, env,),
+                                                 kwargs={"sub_type": "win_heavyload"})
```

Comment:
Could you set it in cfg file,
and some params is needed:
install_cmd, config_cmd, details pls check win_heavyload test case.